smarkets/streaming_api/framing.py

```diff
@@ -1,79 +1,81 @@
 from __future__ import absolute_import, division, print_function, unicode_literals
 
-from six import indexbytes, int2byte
+from six import int2byte
 
 
 def frame_encode(payload):
     """
-    :type payload: byte string
-    :rtype: byte string
+    :type payload: byte string or bytearray
+    :rtype: bytearray
     """
     byte_count = len(payload)
     header = uleb128_encode(byte_count)
+    padding = b'\x00' * max(0, 4 - len(header) - byte_count)  # Minimum frame size is 4 bytes
+    frame = bytearray()
+    frame += header
+    frame += payload
+    frame += padding
+    return frame
 
-    # Minimum frame size is 4 bytes
-    padding = b'\x00' * max(0, 4 - len(header) - len(payload))
 
-    return header + payload + padding
-
-
-def frame_decode_all(string):
+def frame_decode_all(byte_array):
     """
-    :type string: byte string
-    :rtype: tuple of (list of payloads) and remaining string
+    :type byte_array: bytearray
+    :rtype: tuple (list of bytearrays, remaining bytearray)
     """
```

Comment:
@AdamStelmaszczyk I thought we were leaving this as `bytes` in the documentation?